doc/nrf/ug_contribution.rst

```diff
@@ -0,0 +1,142 @@
+The following steps show a typical workflow:
+
+1. :ref:`dm-wf-fork`.
+   The following substeps summarize the most important commands to start working with a fork of the repository:
+
+   a. Fork the fw-nrfconnect-nrf repository to your personal account on GitHub.
+      To do so, click the `fork button on the fw-nrfconnect-nrf repository page`_ (upper right corner).
+
+   #. On your computer, change into the :file:`nrf` folder that was created when you :ref:`obtained the code <cloning_the_repositories_win>`::
+
+         cd ncs/nrf
+
+      Rename the default remote that points to the `upstream nRF Connect SDK repository <fw-nrfconnect-nrf_>`_ from ``origin`` to ``ncs``::
+
+         git remote rename origin ncs
+
+      Let Git know about the fork that you just created and name it ``origin``.
+      Replace *your_github_id* with your GitHub user name.
+
+      .. parsed-literal::
+         :class: highlight
+
+         git remote add origin https\://github.com/*your_github_id*/fw-nrfconnect-nrf
+
+      Finally, verify the remote repositories::
+
+         git remote -v
+
+      The output should look similar to this:
+
+      .. parsed-literal::
+         :class: highlight
+
+         origin   https\://github.com/*your_github_id*/fw-nrfconnect-nrf (fetch)
+         origin   https\://github.com/*your_github_id*/fw-nrfconnect-nrf (push)
+         ncs      https\://github.com/NordicPlayground/fw-nrfconnect-nrf (fetch)
+         ncs      https\://github.com/NordicPlayground/fw-nrfconnect-nrf (push)
+
+#. Create a topic branch (based on the master branch) for your work.
+   Replace *branch_name* with the name that you want to use for the local branch.
+
+   .. parsed-literal::
+      :class: highlight
+
+      git checkout -b *branch_name* ncs/master
+
+   If you are addressing a specific issue, we suggest including the issue number in the branch name.
+
+#. Make changes, test locally, change, test, test again, ...
+   Check out the *sanitycheck* section in Zephyr's :ref:`zephyr:Contribution Tools` documentation as well.
+
+#. When things look good, start the pull request process by adding your changed files.
+   Replace *files* with the file paths of the files that you want to add (see the Git documentation for other options).
+
+   .. parsed-literal::
+      :class: highlight
+
+      git add *files*
+
+   You can see files that are not yet staged with the following command::
+
+     git status
+
+#. Verify that the changes to be committed look as you expect::
+
+     git diff --cached
+
+#. Commit the changes to your local repository::
+
+     git commit -s
+
+   The ``-s`` option automatically adds your ``Signed-off-by:`` to your commit message.
+   This signature indicates that you agree with the :ref:`zephyr:DCO`.
+   Without this line, your commit will be rejected.
+   See the *Commit Guidelines* section in Zephyr's :ref:`zephyr:Contribution workflow` for specific guidelines for writing your commit messages.
```

Comment:
We don't have commit guidelines for NCS - does it make sense to link to Zephyr's?